### DISCUSSION TITLE
Restriction on type not applying when used within a mapping

### DISCUSSION THREAD

**Discussion by aka-mj** [apple/pkl#348] [Q&A] [answered]
I have the following

```
typealias Lang = String(matches(Regex(#"^[a-z]{2}$"#)))

m = new Mapping<String, Lang> {
	["0"] = "en"
	["1"] = "de"
	["2"] = "fr"
	["3"] = "e"
}
```

I'd expect an error because line `["3"] = "e"` value doesn't match the regex for type `Lang`, but pkl produces the following:

```
{
  "m": {
    "0": "en",
    "1": "de",
    "2": "fr",
    "3": "e"
  }
}

```

**Comment by odenix** [comment, marked as answer]
It's a known issue.
Mapping entries are only type-checked if the mapping is assigned to a typed property:
```
m: Mapping<String, Lang> = new {...}
```
The same holds for listing elements.
